handler/producer/sequence.go

```diff
@@ -87,16 +121,40 @@ func pipe(req *http.Request, rt []Roundtrip, kind string) chan *Result {
 	return result
 }
 
-func pipeResult(ctx context.Context, req *http.Request, results chan *Result, rt Roundtrip) {
+func pipeResults(results, rch2 chan *Result) {
+	defer close(results)
+	for r := range rch2 {
+		results <- r
+	}
+}
+
+func produceAndPipeResults(ctx context.Context, req *http.Request, results chan *Result, rt Roundtrip, additionalChs *sync.Map) {
 	outreq := req.WithContext(ctx)
 	defer close(results)
+	rs := rt.Produce(outreq, additionalChs)
+
+	k := fmt.Sprintf("%v", rt.Names())
+	var additional []chan *Result
+	if val, ok := additionalChs.Load(k); ok {
+		additional = val.([]chan *Result)
+	}
+	for _, ach := range additional {
+		defer close(ach)
+	}
 
-	for r := range rt.Produce(outreq) {
+	for r := range rs {
 		select {
 		case <-ctx.Done():
-			results <- &Result{Err: ctx.Err()}
+			e := &Result{Err: ctx.Err()}
+			results <- e
+			for _, ach := range additional {
```

Comment:
Do not write here, fast exit. The channel reader could have hang up already so this could be blocking. Also no emergency close, the reader will stop reading anyways as the should listen to `ctx.Done`.

Reply:
The parent `results` gets just one err entry and gets closed due to return and defer. No one will read then all the other entries (blocking).